src/java.base/share/classes/java/text/DigitList.java

```diff
@@ -508,94 +511,50 @@ private boolean shouldRoundUp(int maximumDigits,
 
             switch(roundingMode) {
             case UP:
-                for (int i=maximumDigits; i<count; ++i) {
-                    if (digits[i] != '0') {
-                        return true;
-                    }
-                }
-                break;
+                return non0AfterIndex(maximumDigits);
             case DOWN:
                 break;
             case CEILING:
-                for (int i=maximumDigits; i<count; ++i) {
-                    if (digits[i] != '0') {
-                        return !isNegative;
-                    }
-                }
-                break;
             case FLOOR:
-                for (int i=maximumDigits; i<count; ++i) {
-                    if (digits[i] != '0') {
-                        return isNegative;
-                    }
+                if (non0AfterIndex(maximumDigits)) {
+                    return (isNegative && roundingMode == RoundingMode.FLOOR)
+                            || (!isNegative && roundingMode == RoundingMode.CEILING);
```

Comment:
`roundingMode` is checked against `FLOOR`/`CEILING` twice. I don't see the need of bundling them up.

Reply:
Thank you for the review. Addressed this, and your other comments that I did not explicitly respond to.

Reply:
I'd think this separate one-liner for each case would be simpler and more readable:
```
    case CEILING:
        return nonZeroAfterIndex(maximumDigits) && !isNegative;
    case FLOOR:
        return nonZeroAfterIndex(maximumDigits) && isNegative;
```